src/Tribe/Views/V2/Utils/Separators.php

```diff
@@ -77,4 +77,67 @@ public static function should_have_time( $events, $event ) {
 
 		return isset( $start_hours_ids[ $event_id ] );
 	}
+
+	/**
+	 * Determines if a given event from a list of events should have a type separator
+	 * for the day view template structure.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  array       $events WP_Post or numeric ID for events.
+	 * @param  WP_Post|int $event  Event we want to determine.
+	 *
+	 * @return boolean
+	 */
+	public static function should_have_type( array $events, \WP_Post $event ) {
+		if ( ! is_array( $events ) ) {
+			return false;
+		}
+
+		$event_id = is_numeric( $event ) ? $event : $event->ID;
+
+		if ( empty( $event->timeslot ) ) {
+			return false;
+		}
+
+		$ids = array_map( static function( $event ) {
+			return absint( is_numeric( $event ) ? $event : $event->ID );
+		}, $events );
```

Comment:
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: a0db3950242d454cb4f8999018a4059e</sub>